lib/services/authentication.go

```diff
@@ -82,23 +76,37 @@ type AuthPreference interface {
 
 // NewAuthPreference is a convenience method to to create AuthPreferenceV2.
 func NewAuthPreference(spec AuthPreferenceSpecV2) (AuthPreference, error) {
-	return &AuthPreferenceV2{
+	pref := AuthPreferenceV2{
 		Kind:    KindClusterAuthPreference,
 		Version: V2,
 		Metadata: Metadata{
 			Name:      MetaNameClusterAuthPreference,
 			Namespace: defaults.Namespace,
 		},
 		Spec: spec,
-	}, nil
+	}
+
+	if err := pref.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &pref, nil
+}
+
+// DefaultAuthPreference returns the default authentication preferences.
+func DefaultAuthPreference() AuthPreference {
+	pref, err := NewAuthPreference(AuthPreferenceSpecV2{})
+	if err != nil {
+		panic(err)
```

Comment:
Let's please not introduce explicit panics anywhere in the code. Just return an error.

Reply:
`Default*` methods do not tend to return an `error` inasmuch as they provide a hard-coded value. If the programmer made a mistake, it seems proper to `panic`.

However, I can take the other route (as in `DefaultClusterConfig` or `DefaultStaticTokens`), skip `NewAuthPreference` with its sanity checks altogether and hard-code the `AuthPreferenceV2` structure as a whole instead. It may even improve performance though the emphasis on data validity/consistncy seemed more beneficial to me.

Reply:
It may be ok to panic upon server initialization (though it's also debatable, in my mind), however I'm thinking about the scenario where the server is running and then someone creates auth preference using this method which, let's suppose, leads to a panic. This will result in the server crashing.

Reply:
Agree with @r0mant on this one. Either return an error or hard code the default value.

Reply:
Done.